### ISSUE TITLE
[BUG]nordictrak T 6.5si can't connect

### ISSUE THREAD

**Issue by sorel123** [cagnulein/qdomyos-zwift#1905]
**Describe the bug**
I've been having issues getting my treadmill to connect. I've installed the latest version of the companion app 3.3.22 and I'm on QZ-fitness 2.16.28. 

It just always shows 0.0 on the tiles. At one point it seemed like the app connected but it wasn't updating speed, just stuck at constant 1.6 k/mh. 

I've uninstalled, reinstalled, rebooted numerous times and haven't been able to get it working. 

**Screenshots**
![image](https://github.com/cagnulein/qdomyos-zwift/assets/13295244/1d4d5c26-6903-424a-afbb-41b013038811)
![image](https://github.com/cagnulein/qdomyos-zwift/assets/13295244/28530137-43ea-48a5-98ae-9177ae22af4b)
![image](https://github.com/cagnulein/qdomyos-zwift/assets/13295244/bd37e558-ad75-436d-a376-06c1201f64c4)

**Smartphone (please complete the following information):**
 - Device: iPhone 13 Pro Max
 - OS: iOS 17
 - Version .2

**Append a debug log**
Will attach separately, can't do it in iOS app I guess.

**Comment by sorel123** [comment]
Picture above was supposed to be having Nordic track t 6.5si v83 selected and ip set.  But chose wrong pic.  Here are debug logs. Thanks!
[2023-12-22_logs.txt](https://github.com/cagnulein/qdomyos-zwift/files/13755242/2023-12-22_logs.txt)
[logcat.txt](https://github.com/cagnulein/qdomyos-zwift/files/13755243/logcat.txt)
[debug-Fri_Dec_22_12_40_46_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/13755244/debug-Fri_Dec_22_12_40_46_2023.log)

**Comment by cagnulein** [comment]
@sorel123 ok the good news is that i can see the messages from the companion but they are empty.
which device did you select in the companion? did you try the c1750 and then restarting the treadmill?

Let me know

**Comment by sorel123** [comment]
> @sorel123 ok the good news is that i can see the messages from the companion but they are empty. which device did you select in the companion? did you try the c1750 and then restarting the treadmill?
> 
> Let me know

I trust tried selecting c1750 treadmill in companion and had same results.  I’ve also tried T 6.5s, t7.5s, and other.  One thing to note my treadmill is the 6.5si model, not sure that means anything.  I seem to remember there was one with a 5-6” and this is an 8-10”.

[debug-Fri_Dec_22_14_08_17_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/13756207/debug-Fri_Dec_22_14_08_17_2023.log)

**Comment by cagnulein** [comment]
did you use the adb method to install the companion? if not are you able to
use adb to collect a adb logcat while the treadmill is running?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno ven 22 dic 2023 alle 20:15 sorel123 ***@***.***> ha
scritto:

**Comment by sorel123** [comment]
I did use adb method.  Here are fresh logcats taken while tread is running. I attached one above too but not sure if tread was running.

[logcat.txt](https://github.com/cagnulein/qdomyos-zwift/files/13756247/logcat.txt)
[logcat.log](https://github.com/cagnulein/qdomyos-zwift/files/13756248/logcat.log)

**Comment by cagnulein** [comment]
ok thanks, from a quick view i don't see any metrics (i'm from the phone)
i will check this again in the next days.
in the meantime you can try to do a factory reset without updating the ifit
app. it should works

let me know

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno ven 22 dic 2023 alle 20:25 sorel123 ***@***.***> ha
scritto:

**Comment by sorel123** [comment]
Thank you, after 2 more resets I was able to get it to read live data.  First reset I entered Wi-Fi info and it auto updated apps, had same issue.  Second time I didn’t enter Wi-Fi and did it through android later.  Now I seem to be getting info into qz app.  Was connecting to peloton as QZ but didn’t get any data or auto follow features.  Will have to read up on how to get that to work.  I’ll try and look at it later.  Thanks a lot for your help!!!

**Comment by cagnulein** [comment]
great! are you on android or ios? did you add your peloton credentials to
qz?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno ven 22 dic 2023 alle 22:27 sorel123 ***@***.***> ha
scritto:

**Comment by cagnulein** [comment]
Are we fine @sorel123 ?

**Comment by sorel123** [comment]
Hi, sorry was going to check back after the holidays. It worked after the wipe but if it updated the next day and it was back to the same issue. I'm guessing I need to block it from updating?

It's such a pain, I tried my parents t6s (no tablet) and it connected first time with no issues.

**Comment by sorel123** [comment]
I reset everything again and I'm again getting data. Last time I did this after iFit updated it stopped working again, I'll see what happens tomorrow. Also QZ doesn't control speed or incline, I can see the data but the +/- in the qz app does nothing. Im on the lastest version of app in iOS TestFlight.

**Comment by cagnulein** [comment]
yes you have to stop ifit to update it. i usually do removing the gateway
field from the network settings in the tablet.

for changing speed and incline: check this
https://github.com/cagnulein/QZCompanionNordictrackTreadmill/wiki/Troubleshooting#i-cant-control-the-speedinclinationresistance-of-my-treadmillbike


Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno lun 8 gen 2024 alle 04:50 sorel123 ***@***.***> ha
scritto:

**Comment by sorel123** [comment]
That seems to work. I set the gateway to itself and it's not updating. I can now get consistent data. Only downside is that old version of iFit takes a full minute to go back to main page. I opened another ticket in the companion app to get the speed/incline mapped. Sorry for all the messages, I really appreciate your help!!!!  I wish I could go back 4 years ago and buy the cheaper t6.5s, your app synced with that instantly.
